Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/parser/PRPAIN201306UV02Parser.java

```diff
@@ -76,4 +77,71 @@ public static String getQueryId(PRPAIN201306UV02 response) {
         }
         return oIIs;
     }
+
+    public static String getSenderHcid(PRPAIN201306UV02 response) {
+        String id = null;
+        if (response != null && response.getSender() != null && response.getSender() != null
+            && response.getSender().getDevice() != null && response.getSender().getDevice().
+            getAsAgent() != null
+            && response.getSender().getDevice().getAsAgent().getValue() != null
+            && response.getSender().getDevice().getAsAgent().getValue().getRepresentedOrganization() != null
+            && response.getSender().getDevice().getAsAgent().getValue().getRepresentedOrganization().getValue().
+            getId() != null
+            && !response.getSender().getDevice().getAsAgent().getValue().getRepresentedOrganization().getValue()
+            .getId().isEmpty()
+            && response.getSender().getDevice().getAsAgent().getValue().getRepresentedOrganization().getValue()
+            .getId().get(0) != null
+            && response.getSender().getDevice().getAsAgent().getValue().getRepresentedOrganization().getValue()
```

Comment:
The NullChecker class has the nullcheck/isempty check for String and List. Here each of them are different objects and hence the check is done here.